test-runner-poc/lib/runner.js

```diff
@@ -2,72 +2,135 @@
 
 const path = require("path");
 const fs = require("fs");
-const utils = require("mocha").utils;
+const MongoClient = require('mongodb').MongoClient;
+const environments = require('../environments');
+const TestConfiguration = require('../config');
+const parseConnectionString = require('../../lib/core/uri_parser');
+const mock = require('mongodb-mock-server');
 
+let mongoClient;
 let filters = [];
 let files = [];
 
 function addFilter(filter) {
-	if (typeof filter !== "function" && typeof filter !== "object") {
-		throw new Error(
-			"Type of filter must either be a function or an object"
-		);
-	}
-	if (
-		typeof filter === "object" &&
-		(!filter.filter || typeof filter.filter !== "function")
-	) {
-		throw new Error("Object filters must have a function named filter");
-	}
-
-	if (typeof filter === "function") {
-		filters.push({ filter: filter });
-	} else {
-		filters.push(filter);
-	}
+  switch (typeof filter) {
```

Comment:
How many filters are currently being used? Are you actually using any filters that are just `function`s? Or are all of them objects lile `mongodb_topology_filter`?

Reply:
They are objects, so we have updated the cases in the switch statement.

Reply:
We have 7 filters

Reply:
If all of the filters are objects, maybe we shouldn't use a `switch` statement here and should instead do something like 
```
if (typeof filter !== 'object') {
  throw new Error('Type of filter must be an object');
}
if (!filter.filter || typeof filter.filter !== 'function') {
  throw new Error('Object filters must have a function named filter');
}
filters.push(filter);
```

Reply:
Changed it to the if statement. Thanks!